### PR TITLE
Move Optimus and SS2 to WDL 1.0

### PR DESCRIPTION
**Pull request by barkasn**

### CODE REVIEW

File: library/tasks/GroupMetricsOutputs.wdl [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
-  String output_name
-  # Runtime
-  String docker = "quay.io/humancellatlas/secondary-analysis-sctools:v0.3.4"
-  Int mem = 3
-  Int cpu = 1
-  Int disk_space = 20 
-  Int preemptible = 5
+  input {
+    Array[File] picard_row_outputs
+    Array[File] picard_row_optional_outputs
```
Comment (jessicaway) [inline library/tasks/GroupMetricsOutputs.wdl]: These no longer need to be optional?

File: library/tasks/HISAT2.wdl [hunk 2]
```diff
@@ -261,7 +267,8 @@ task HISAT2SingleEnd {
 }
 
 task HISAT2InspectIndex {
-  File hisat2_ref
+input {
+File hisat2_ref
```
Comment (jessicaway) [inline library/tasks/HISAT2.wdl]:
```suggestion
  File hisat2_ref
```

File: library/tasks/OptimusInputChecks.wdl [hunk 3]
```diff
@@ -15,8 +19,8 @@ task checkOptimusInput {
     ## Perform checks
     if [[ ! ("${chemistry}" == "tenX_v2" || "${chemistry}" == "tenX_v3") ]]
     then
-	pass="false"
-	echo "ERROR: Invalid value \"${chemistry}\" for input \"chemistry\""
+  pass="false"
```
Comment (jessicaway) [inline library/tasks/OptimusInputChecks.wdl]: indentation looks incorrect now

File: library/tasks/Picard.wdl [hunk 4]
```diff
@@ -80,19 +87,22 @@ task SortBamAndIndex {
...
-  Int disk = ceil(size(aligned_bam, "GiB") + size(genome_ref_fasta, "GiB") + 10)
-  Int preemptible = 5
+    # runtime values
+    String docker ="quay.io/humancellatlas/secondary-analysis-picard:v0.2.2-2.10.10"
+    Int machine_mem_mb = 8250
+    # give the command 1 GiB of overhead
+    Int command_mem_mb = machine_mem_mb - 1000
+    Int cpu = 1
+    # use provided disk number or dynamically size on our own, with 10GiB of additional disk
+    Int disk = ceil(size(aligned_bam, "GiB") + size(genome_ref_fasta, "GiB") + 10)
```
Comment (jessicaway) [inline library/tasks/Picard.wdl]: Should this be below the input block like the other ones that depend on inputs?

File: library/tasks/Picard.wdl [hunk 5]
```diff
@@ -159,21 +169,24 @@ task CollectMultipleMetrics {
...
-  Int disk = ceil(size(aligned_bam, "GiB") + size(ref_flat, "GiB") + size(rrna_intervals, "GiB") + 10)
-  Int preemptible = 5
+    # runtime values
+    String docker = "quay.io/humancellatlas/secondary-analysis-picard:v0.2.2-2.10.10"
+    Int machine_mem_mb = 3850
+    # give the command 500 MiB of overhead
+    Int command_mem_mb = machine_mem_mb - 500
+    Int cpu = 1
+    # use provided disk number or dynamically size on our own, with 10GiB of additional disk
+    Int disk = ceil(size(aligned_bam, "GiB") + size(ref_flat, "GiB") + size(rrna_intervals, "GiB") + 10)
```
Comment (jessicaway) [inline library/tasks/Picard.wdl]: Same comment as above

File: library/tasks/Picard.wdl [hunk 6]
```diff
@@ -222,18 +235,21 @@ task CollectRnaMetrics {
...
-  Int disk = ceil(size(aligned_bam, "GiB") + 10)
-  Int preemptible = 5
+    # runtime values
+    String docker = "quay.io/humancellatlas/secondary-analysis-picard:v0.2.2-2.10.10"
+    Int machine_mem_mb = 8250
+    # give the command 1 GiB of overhead
+    Int command_mem_mb = machine_mem_mb - 1000
+    Int cpu = 2
+    # use provided disk number or dynamically size on our own, with 10GiB of additional disk
+    Int disk = ceil(size(aligned_bam, "GiB") + 10)
```
Comment (jessicaway) [inline library/tasks/Picard.wdl]: same

File: library/tasks/ScatterBam.wdl [hunk 7]
```diff
@@ -1,18 +1,22 @@
...
 
-  command <<<
+  command {
     mkdir scattered_bams
     java -Xms7g -jar /usr/picard/picard.jar \
       SplitSamByNumberOfReads \
       INPUT=${bam_to_scatter} \
       SPLIT_TO_N_FILES=${scatter_width} \
-      OUT_PREFIX=$(basename ${bam_to_scatter} .bam)_split \
+      OUT_PREFIX=${basename(bam_to_scatter, '.bam')}_split \
```
Comment (jessicaway) [inline library/tasks/ScatterBam.wdl]: This looks incorrect

File: test/optimus/4kpbmc/ValidateOptimus.wdl [hunk 8]
```diff
@@ -90,10 +96,12 @@ task ValidateBam {
 }
 
 task ValidateLoom {
-    File loom_file
-    String expected_loom_file_checksum
+    input {
+        File loom_file
```
Comment (jessicaway) [inline test/optimus/4kpbmc/ValidateOptimus.wdl]: does this need to be optional?

File: library/tasks/GroupMetricsOutputs.wdl (hunk 1 above)
Comment (barkasn) [reply library/tasks/GroupMetricsOutputs.wdl]: I don't know, but I would think its a different ticket to change that. I think it should be in inputs though. I'll fix

File: library/tasks/Picard.wdl (hunk 4 above)
Comment (barkasn) [reply library/tasks/Picard.wdl]: I don't know, @pshapiro4broad ? that's something Bo's team did. I thihg everything should be under input, but I might be wrong. I'd like Phils input

File: library/tasks/ScatterBam.wdl (hunk 7 above)
Comment (barkasn) [reply library/tasks/ScatterBam.wdl]: Yes I think they changed that. Not sure why. Seems to be wroking fine. I think they are usintfuhe wdl basename function. @pshapiro4broad  ??

File: test/optimus/4kpbmc/ValidateOptimus.wdl (hunk 8 above)
Comment (barkasn) [reply test/optimus/4kpbmc/ValidateOptimus.wdl]: Yes well spotted, thank you. I really I forgot to make sure that the 4kpmbc test also runs correctly, will amend

File: library/tasks/Picard.wdl (hunk 4 above)
Comment (jessicaway) [reply library/tasks/Picard.wdl]: I think everything in the input block could be nice, that way you can easily override it from the inputs json if you need to change it

File: library/tasks/GroupMetricsOutputs.wdl (hunk 1 above)
Comment (barkasn) [reply library/tasks/GroupMetricsOutputs.wdl]: Oh actually I see what you mean. These can't be optional with WDL 1.0 --this is the issue that Denis helped fix. In practice they never are omitted so this is ok.

**Pull request merged**